script/fix_claim_reference.rb

```diff
@@ -0,0 +1,16 @@
+require File.expand_path('../../config/environment',  __FILE__)
+
+
+Card::Auth.as_bot do
+
+  claim_cards = Card.search :type_id=>Card::ClaimID,:not=>{:right_plus=>[Card[:source].name,{:refer_to=>{:type_id=>Card::SourceID} }]}
+  claim_cards.each do |card|
+    claim_source = card.fetch :trait=>:source
+    if claim_source
+      puts claim_source.name
+      claim_source.update_references
+      claim_source.expire_structuree_references
```

Comment:
I think expire_structuree_references only applies to cards that are structures (in fact, that method and several others in templating.rb should probably be in right/structure.rb).  So I doubt this line does everything.